slackevents/inner_events.go

```diff
@@ -258,9 +258,16 @@ type MessageEvent struct {
 	UserTeam   string `json:"user_team,omitempty"`
 	SourceTeam string `json:"source_team,omitempty"`
 
+	// When we get a 'message' event with no subtype, i.e. telling us about a new
+	// message, the message information is stored at the top level. But when we get
+	// a 'message_changed' event, the message information is stored in
+	// the Message property. This is really hard to represent nicely in Go, so we use
+	// a custom JSON unmarshaller to populate the Message field in both cases.
+	Message *slack.Msg `json:"message,omitempty"`
+	// Root is set if the SubType is `thread_broadcast`.
+	Root *slack.Msg `json:"root,omitempty"`
 	// Edited Message
-	Message         *slack.Message `json:"message,omitempty"`
-	PreviousMessage *slack.Message `json:"previous_message,omitempty"`
+	PreviousMessage *slack.Msg `json:"previous_message,omitempty"`
```

Comment:
Switched these from `slack.Message` as `slack.Message` is a thin wrapper which includes previous_message and root. We have these in our 'MessageEvent' struct, so don't want to also include them inside the struct as it's misleading.